### ISSUE TITLE
Filters aren't mean-insensitive if low-passed

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#176]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/176) by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

@{557058:6b1a4860-1eb7-4caa-9340-5e3c97a89faf} and @{557058:9524f6a0-ef9f-4466-b913-d04e51ff9093} discovered that using `f_3db` as a low-pass filter \(or presumably also `fmax`\) creates a problem: filters filtered in this way stop being DC-insentive. They apparently introduce a DC term into the filter. It should be removed. Paul is about to add a note/example code about this.

**Comment by joefowler** [comment]
**Original comment by Paul Szypryt (Bitbucket: [pszypryt](https://bitbucket.org/pszypryt), GitHub: [pszypryt](https://github.com/pszypryt)).**

----------------------------------------

Here is a minimal argument example that shows the mean of the filter becoming non-zero when a f\_3db cutoff is used.

‌

‌

```python
import mass
import numpy as np, pylab as plt
import sys
import os


nSamples = 100
nPresamples = 50

fake_pulse_like = np.append(np.zeros(nPresamples), np.random.randn(nSamples-nPresamples)+1)
fake_deriv_like = np.append(np.zeros(nPresamples), np.random.randn(nSamples-nPresamples)+1)

fake_model = np.column_stack((fake_pulse_like, fake_deriv_like))

fakeNoise=np.random.randn(nSamples)
whitener=None
sample_time= 6.72e-6

fake_f_ATSF = mass.ArrivalTimeSafeFilter(fake_model, nPresamples, fakeNoise, whitener, sample_time, np.max(fake_pulse_like))

fake_f_ATSF.compute()
print("Mean computed without f_3dB cutoff: {}".format(np.mean(fake_f_ATSF.filt_noconst)))

fake_f_ATSF.compute(f_3db=1e4)
print("Mean computed with 1e4 f_3dB cutoff: {}".format(np.mean(fake_f_ATSF.filt_noconst)))
```

‌

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Handle band limits on ArrivalTimeSafeFilters differently

This will ensure that a DC-insensitive ATS Filter will remain so even if a band limit is imposed. Fixes issue #176. HOWEVER, beware that it actually changes how filters are computed and might well yield different results, for better or for worse.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Removing milestone: v0.7 (automated comment)
